OpenRA.Game/Game.cs

```diff
@@ -279,6 +269,25 @@ internal static void Initialize(Arguments args)
 				}
 			}
 
+			GeoIP.Initialize();
+
+			try
+			{
+				if (!Game.Settings.Server.DiscoverNatDevices)
+					Game.Settings.Server.AllowPortForward = false;
+				else
+				{
+					UPnP.DiscoverNatDevices(Settings.Server.NatDiscoveryTimeout).ConfigureAwait(false);
```

Comment:
If we don't `await` this, then there's no guarantee this has completed! This means calls to forward ports later would fail since devices won't have been discovered yet.

You can kick the tyres on the task by doing
`var deviceDiscovery = UPnP.DiscoverNatDevices(Settings.Server.NatDiscoveryTimeout);` back up top - to get it going early. This means discovery will be running in the background whilst the rest of loading takes place.

You can then do `await deviceDiscovery;` here to ensure it has completed.

Your try-catch should be around the awaited task, as that's where any exceptions would be thrown.